localstack/utils/bootstrap.py

```diff
@@ -445,6 +445,23 @@ def run(self):
                 workdir=self.workdir,
                 privileged=self.privileged,
             )
+
+        try:
+            volume_mounts = self._get_mount_volumes()
+            try:
+                return _run_container(volume_mounts)
+            except ContainerException as e:
+                err_str = str(e) + str(e.__cause__)
+                if "Mounts denied" in err_str:
+                    # This can happen if Docker doesn't have access to the LocalStack volume directory.
+                    # We remove the volume dir mount from the list, and then attempt to run the container again.
+                    LOG.warning(
+                        "Error mounting volumes in Docker - attempting again without LocalStack volume mount"
+                    )
+                    volume_mounts = [vol for vol in volume_mounts if vol[1] != DEFAULT_VOLUME_DIR]
+                    DOCKER_CLIENT.remove_container(self.name, force=True)
+                    return _run_container(volume_mounts)
+                raise
```

Comment:
would be great if we could get a second pair of :eyes: here from @dfangl

Reply:
I am also not a fan of this, more or less silent, automatic fallback here. While we will see there is an error if the customers send us the logs due to the warning (the warning should definitely include the error itself here!), the fallback might not be clear to the user as the warning is easily missed. This can lead to increased support loads over a immediate failure with (ideally clear) error messages.

In my opinion, for such a case, we should print the complete error returned by the docker daemon (the whole "this directory is not shared with the VM" thing), and abort. As a user, I would like it better for the software to fail hard at the first time with a clear error message, than to get unexpected behavior later (no persistence working for example), after the warning is already burried in the logs.

Reply:
Fair points! Will revert these changes for now (the other part of the PR related to `config.dirs`/`OVERRIDE_IN_DOCKER` is the more critical one for now anyway), we can see over time if the volume mount permissions actually cause issues with our users. 

On second thought, agree with both of you that silently not mounting a LocalStack volume dir may be problematic. 👍 In the particular example above, it can be easily worked around by setting `LOCALSTACK_VOLUME_DIR`, e.g.:
```
$ docker run -v /var/run/docker.sock:/var/run/docker.sock -it python:3 bash
root@ca6c629167ed:/# pip install docker localstack
...
root@ca6c629167ed:/# LOCALSTACK_VOLUME_DIR=/tmp DEBUG=1 localstack start
```